### PR TITLE
refactor top perfs filter

### PR DESCRIPTION
**Pull request by pierrelouis-c**
### ❓ Context

- **Impacted projects**:  @ledgerhq/live-common, live-mobile
  
- **Linked resource(s)**: [LIVE-2392]

We needed to make that refactor to be able to make a search on the top 100 and to move the filter logic out of the Market component.

### ✅ Checklist

- [ ] **Test coverage**: _Did you write any tests to cover the changes introduced by this pull request?_
- [ ] **Atomic delivery**: _Is this pull request standalone? In order words, does it depend on nothing else?_
- [ ] **No breaking changes**: _Does this pull request contain breaking changes of any kind? If so, please explain why._

### 📸 Demo

https://user-images.githubusercontent.com/29443638/169496588-bf5fbf67-266c-402e-bf5c-ac6b198cdf60.mp4

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-2392]: https://ledgerhq.atlassian.net/browse/LIVE-2392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/ledger-live-common/src/market/api/api.ts [hunk 1]
```diff
@@ -137,11 +140,25 @@ async function listPaginated({
...
+      .sort(
+        (a, b) =>
+          b[`price_change_percentage_${range}_in_currency`] -
+          a[`price_change_percentage_${range}_in_currency`]
+      )
+      .filter((currency) => {
+        if (!search) return true;
+        const match = `${currency.symbol}|${currency.name}`;
+        return match.toLowerCase().includes(search.toLowerCase());
+      });
```
Comment (gre) [inline libs/ledger-live-common/src/market/api/api.ts:159]: I think moving the filter BEFORE the sort helps the performance.

Comment (pierrelouis-c) [reply libs/ledger-live-common/src/market/api/api.ts]: Thank you for your feedback. I made the change :)

**Review by valpinkman — APPROVED** [review]

File: libs/ledger-live-common/src/market/api/api.ts [hunk 2]
```diff
@@ -137,11 +140,25 @@ async function listPaginated({
...
 
-  const { data } = await network({
+  let { data } = await network({
     method: "GET",
     url,
   });
 
+  if (top100) {
+    data = data
+      .filter((currency) => {
```
Comment (pierrelouis-c) [inline libs/ledger-live-common/src/market/api/api.ts:150]: add comment to details the code

**Review by valpinkman — APPROVED** [review]

**Review by thomasrogerlux — APPROVED** [review]
Test review ok!

**Pull request merged**